StringCalculator/StringCalculator.cs

```diff
@@ -0,0 +1,98 @@
+﻿namespace StringCalculator
+{
+    public class StringCalculator
+    {
+        private const int maxNumberToAdd = 1000;
+
+        public int Add(string numbersGiven)
+        {
+            if (string.IsNullOrEmpty(numbersGiven))
+            {
+                return 0;
+            }
+
+            string[] splitNumbersToAdd = SplitStringWithDefaultOrCustomSeperators(numbersGiven);
+
+            ValidateNumbersForNegatives(splitNumbersToAdd);
+
+            string[] splitAndFilteredNumbers = FilterNumbersToSumUnderLimit(splitNumbersToAdd);
+
+            int finalResult = CalculateSplitNumberAddition(splitAndFilteredNumbers);
+
+            return finalResult;
+        }
+
+        private string[] SplitStringWithDefaultOrCustomSeperators(string numbersToSplit)
+        {
+            string numbersToCheckForSeperstors = numbersToSplit;
+
+            string[] seperators = new string[] { ",", "\\n" };
+
+            if (numbersToCheckForSeperstors.StartsWith("//"))
+            {
+                string seperatorFromInput = numbersToSplit[2..numbersToSplit.IndexOf("\\n")];
+                seperators = seperatorFromInput.Split(new string[] { "[", "]" }, StringSplitOptions.RemoveEmptyEntries);
+
+                numbersToCheckForSeperstors = numbersToSplit.Substring(numbersToSplit.IndexOf("\\n") + 2);
+            }
+
+            string[] numbersSplit = numbersToCheckForSeperstors.Split(seperators, StringSplitOptions.None);
```

Comment:
Does it make sense from a design perspective to allow for a tonne of empty strings to move around your system? It means for additional checks, more complicated code, and more chance for error. It's equatable to passing nulls around the system.